src/Cassandra.IntegrationTests/Core/PoolShortTests.cs

```diff
@@ -214,5 +214,40 @@ public void Connect_With_Ssl_Test()
                 });
             }
         }
+
+        [Test]
+        [TestCase(ProtocolVersion.MaxSupported, 1, 2)]
+        [TestCase(ProtocolVersion.V2, 2, 8)]
+        public void PoolingOptions_Create_Based_On_Protocol(ProtocolVersion protocolVersion, 
+            int coreConnectionLength, int maxConnectionLength)
+        {
+            var sCluster = SimulacronCluster.CreateNew(new SimulacronOptions());
+            var options1 = PoolingOptions.Create(protocolVersion);
+            using(var cluster = Cluster.Builder()
+                                       .AddContactPoint(sCluster.InitialContactPoint)
+                                       .WithPoolingOptions(options1)
+                                       .Build())
+            {
+                var session = (Session) cluster.Connect();
+                var allHosts = cluster.AllHosts();
+                var host = allHosts.First();
+                var pool = session.GetOrCreateConnectionPool(host, HostDistance.Local);
+
+                TestHelper.WaitUntil(() =>
+                    pool.OpenConnections == coreConnectionLength);
+                var ports = sCluster.GetConnectedPorts();
+                //coreConnectionLength + 1 (the control connection) 
+                Assert.AreEqual(coreConnectionLength + 1, ports.Count);
+                
+                TestHelper.WaitUntil(() =>
+                    {
+                        //Force pool to resize connections to max
+                        pool.ConsiderResizingPool(int.MaxValue);
```

Comment:
We should avoid using internal methods in integration tests.
Using core connections to test that the settings are being honored is enough for me.